### DISCUSSION TITLE
Best practice for communicating dependencies (in python)

### DISCUSSION THREAD

**Discussion by WatKey** [community/community#21522] [Other Feature Feedback, Questions, & Ideas]
<p>Hi everybody,</p>
<p>I am about to start a project in python which uses a couple dependecies and I was wondering if there is best practice when it comes to communicating and handling these dependencies. Are they in general just written down in a file or is the best approach to write a script that sets up the needed environment and the dependencies?</p>
<p>Thank you for your help</p>

**Comment by fire-eggs** [comment]
<p>A couple of possibilities: Create a requirements.txt file, or use pipenv to manage a virtual environment.</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/watkey">@watkey</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>When you develop a python project is necessary to use some sort of virtual environment. I wrote a couple weeks ago some notes about venv and reproducibility to mantain and share a pyhton project: <a href="https://wabri.github.io/post/pip_environments/" rel="nofollow noopener">pip venv</a>.</p>
<p>Hope this help, ask if you have more questions.</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by SH4FI-DEV** [comment]
<p>What is pip venv and why do we need this ? How to vet started with this?</p>

**Comment by Wabri** [comment]
<p>Venv is a pyhton package that can create a virtual environment where you can install and do staff without affect your local environment, this can be use to create reproducibility for your application.</p>
<p>To get started with this you can read my post <a href="https://wabri.github.io/post/pip_environments/" rel="nofollow noopener">pip venv</a> or use the official docs <a href="https://docs.python.org/3/library/venv.html" rel="nofollow noopener">here</a>.</p>
